src/interpreter/plugin/FinancialPlugin.ts

```diff
@@ -464,15 +464,16 @@ export class FinancialPlugin extends FunctionPlugin implements FunctionPluginTyp
   }
 
   public rate(ast: ProcedureAst, state: InterpreterState): InterpreterValue {
+    // Newton's method: https://en.wikipedia.org/wiki/Newton%27s_method
     return this.runFunction(ast.args, state, this.metadata('RATE'),
       (periods, payment, present, future, type, guess) => {
         if (guess <= -1) {
           return new CellError(ErrorType.VALUE)
         }
 
-        const epsMax = 1e-10
+        const epsMax = 1e-7
 
-        const iterMax = 20
+        const iterMax = 50
```

Comment:
> the spec mentions that the maximum number iterative calculations of cyclic references should be configurable by the host app (HOST-ITERATION-STEPS), but that is a different number than used here, right?

I believe it's a parameter for the algorithm that finds cycles in the dependency graph. It has nothing to do with the `RATE` function.

ODFF specifies [here](https://docs.oasis-open.org/office/OpenDocument/v1.3/os/part4-formula/OpenDocument-v1.3-os-part4-formula.html#RATE) what this function should calculate but not what algorithm should be used let alone the parameters of this algorithm.

In my opinion the value of this parameter is implementation-specific. [This software](https://support.microsoft.com/en-us/office/rate-function-9f665657-4a7e-4bb7-a030-83fc59e748ce) claims that it limits the number of iteration to `20`, but we cannot really compare it, because their implementation is certainly different. One could write this function is such a way that it would perform 2 steps per 1 iteration of a loop and than it would converge in less iterations but it would still take the same time to complete.

> If we hardcode a value that is unreasonably big, it might become a problem. However, this function does not look particularly expensive, so it shouldn't be a problem, would you agree?

I checked and even with `iterMax = 5000` **our implementation** of this algorithm completes in ~10ms on my PC.

Reply:
Thanks!